src/main/java/bio/terra/pdao/bigquery/BigQueryPdao.java

```diff
@@ -560,10 +579,51 @@ CROSS JOIN UNNEST(S.refColumnName) AS refColumnName
         return refIdArray;
     }
 
+    public Set<String> getRowIds(Dataset dataset, String tableName, String projectId, BigQueryProject bigQueryProject) {
+        String softDeleteTableName = prefixSoftDeleteTableName(tableName);
+        String table = projectId + "." + prefixName(dataset.getName()) + "." + tableName;
+        String softDeleteTable = projectId + "." + prefixName(dataset.getName()) + "." + softDeleteTableName;
+        StringBuilder builder = new StringBuilder();
+        builder.append("WITH not_soft_deleted AS ((SELECT ")
+            .append(PDAO_ROW_ID_COLUMN)
+            .append(" FROM `")
+            .append(table)
+            .append("`) EXCEPT DISTINCT ( SELECT ")
+            .append(PDAO_ROW_ID_COLUMN)
+            .append(" FROM `")
+            .append(softDeleteTable)
+            .append("` )) SELECT ")
+            .append(PDAO_ROW_ID_COLUMN)
+            .append(" FROM not_soft_deleted LEFT JOIN `")
```

Comment:
I'm missing two things about the query.
1. You can get the database system to give you a unique set of row ids, so you don't need to use a Set to eliminate duplicates. It doesn't look like you use the unique properties of the Set elsewhere.
2. What is the use of the LEFT JOIN? LEFT JOIN will return all of the row_ids of `not_soft_deleted` and either NULL or the the row_id of the `table` table. Since the select list is `PDAO_ROW_ID_COLUMN`, I think the effect is to return the row ids of `not_soft_deleted`.  I'm guessing that is not what you were trying to do.